docs/guide/features.md

```diff
@@ -131,6 +131,21 @@ Alternatively, you can add `vite/client` to `compilerOptions.types` inside `tsco
 }
 ```
 
+::: warning `types` is an explicit list.
+When `types` is undefined, TypeScript implicitly includes all your installed `@types` modules. If you explicitly declare `types`, that implicit include will be overridden.
+
+To preserve any types your project depends on, explicitly include them in the list. For instance, if you have `@types/colab` installed, your `compilerOptions` should look like this:
+
+```json
+{
+  "compilerOptions": {
+    "types": ["colab", "vite/client"]
+  }
+}
+```
+
+:::
```

Comment:
Could we add a short link to https://www.typescriptlang.org/tsconfig#types explaining the caveats of it instead? e.g.  a one line paragraph:

> Note that [compilerOptions.types](https://www.typescriptlang.org/tsconfig#types) will override any existing global types.

Since our templates aren't using this config by default (due to the same reasons), so we can save some space here.

Reply:
[Is that better?](https://deploy-preview-14861--vite-docs-main.netlify.app/guide/features.html#client-types)

Reply:
> Be sure to include any @types imports, as [they will be overriden when you specify "types"](https://www.typescriptlang.org/tsconfig#types).

This isn't accurate and would give people the impression that they need to include every `@types/whatever` package, rather than just the few that define global typings.

I might suggest:

> You may need to list other dependencies here if they define global types

Reply:
TypeScript has several mechanisms for automatically importing type declaration files, allowing you to get type hints even without explicitly specifying them in `compilerOptions.types`.

The TypeScript compiler will automatically find the `@types` package installed in the project and automatically import the corresponding type declaration file based on the import statements you use in your code. If you use `import _ from 'lodash'` in your code, the compiler will automatically find and import the type definitions in `@types/lodash` , even though `lodash` is not added to the types field of compilerOptions you can still get type hints.

Using the automatic import mechanism can reduce the tedious configuration and comply with the conventions of the TypeScript community. For `@types` packages that comply with the naming convention, the automatic introduction mechanism can easily provide type checking and intelligent sensing support for your code.

The automatic import mechanism applies to most common JavaScript libraries, because they usually have corresponding `@types` packages, and these `@types` packages have been named according to the convention.

However, for some unconventional, customized or non-conforming type declaration files, the automatic import mechanism may not work properly. In this case, manually declaring the types in `compilerOptions.types` is more reliable.

The advantage of manually declaring types is that you can explicitly specify the type declaration files that the project depends on, and control the version and source of the type definition. This improves code readability and maintainability and reduces unnecessary dependencies and confusion.

Therefore, for common JavaScript libraries, it is recommended to use the automatic import mechanism. For unconventional, custom, or nonconforming type declaration files, it is recommended to manually declare them in the types field. On a case-by-case basis, the appropriate approach can be chosen based on the needs and preferences of the project.

By the way, this is a knowledge point of typescript. Vite does not need to explain it.